src/v/debug_bundle/types.h

```diff
@@ -74,6 +74,9 @@ struct partition_selection {
     model::topic_namespace tn;
     absl::btree_set<model::partition_id> partitions;
 
+    static std::optional<partition_selection>
```

Comment:
Perhaps this should be `result<partition_selection>`, we'd have to include `error.h` from `types.h`.